src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/InvokeRestMethodCommand.CoreClr.cs

```diff
@@ -95,6 +95,12 @@ internal override void ProcessResponse(HttpResponseMessage response)
                 {
                     StreamHelper.SaveStreamToFile(responseStream, QualifiedOutFile, this);
                 }
+
+                if (!String.IsNullOrEmpty(ResponseHeadersVariable))
+                {
+                    PSVariableIntrinsics vi = SessionState.PSVariable;
+                    vi.Set(ResponseHeadersVariable, WebResponseHelper.GetHeadersDictionary(response));
```

Comment:
Should we return ReadOnlyDictionary?
Does it make sense to return an _entire_ response?

Reply:
I have seen cases where users are taking the returned dictionary, updating it, and then supplying back to further calls. I don't see much benefit. Also, `System.Net.Http.Headers.HttpResponseHeaders` is not read-only.

```powershell
$Res = invoke-webrequest https://google.com
$Res.BaseResponse.Headers.TryAddWithoutValidation('testy','lala')
$Res.BaseResponse.Headers.GetValues('testy')
```
results with `lala`

As for the returning entire response, that would be somewhat pointless with `Invoke-RestMethod`, IMO. If the user needs the full response object, they can use `Invoke-WebRequest`. This is just adding the ability to get the response headers which many API's use to provide information along with the JSON/XML response. But, a separate PR could be made to return the response object if that is something people really want.